cwms-data-api/src/main/java/cwms/cda/api/LevelsController.java

```diff
@@ -268,6 +298,14 @@ public void getAll(Context ctx) {
 
             if ("2".equals(version)) {
 
+                if (unit == null){
+                    // The dao currently only supports SI.
+                    unit = UnitSystem.SI.getValue();
+                }
+                if(!UnitSystem.SI.getValue().equals(unit)) {
+                    throw new IllegalArgumentException("Levels Version 2 currently only supports SI");
```

Comment:
Doesn't this also support just setting the unit? I swear I corrected some behavior for Ruth to get %, maybe it's a documentation thing though, the levels operate just like the timeseries as the units can be anything not just a length.

Reply:
The getAll is querying against a AV_LOCATION_LEVEL and feeding the rows into the older JDom* classes. 
The Dao code for that method is clearly based on:
https://bitbucket.hecdev.net/projects/CWMS/repos/hec-cwms-data-access/browse/hec-db-jdbc/src/main/java/wcds/dbi/oracle/cwms/CwmsLevelJdbcDao.java

It looks to me like your changes were for calls from getOne for a single Level.  But maybe there are other pl/sql procs we should call instead of hitting the view.

Reply:
Well, that structure of this is fine, so we'll go for it for now. But someone is likely going to complain about not being able to set their preferred units.